### PR TITLE
ILS-249 Add remove field to IBMLicensingDefinition

### PR DESCRIPTION
**Pull request by alicenoknow**
This PR adds `remove` field to `IBMLicensingDefinition`. It should work similar to set field but for removing the annotations that may be already present in pod.

### CODE REVIEW

**Review by tojoos — APPROVED** [review]
Looks good 👍

File: api/v1/ibmlicensingdefinition_types.go [hunk 1]
```diff
@@ -44,6 +44,9 @@ type IBMLicensingDefinitionSpec struct {
 
 	// List of annotations that matched pod would be extended
 	Set map[string]string `json:"set"`
+
+	// List of annotations that will be removed from matched pod
```
Comment (kflorianski-ibm) [inline api/v1/ibmlicensingdefinition_types.go:48]: `List of annotations (keys)` or `List of annotation keys` maybe?

File: api/v1/ibmlicensingdefinition_types.go [hunk 2]
```diff
@@ -44,6 +44,9 @@ type IBMLicensingDefinitionSpec struct {
...
+	Remove []string `json:"remove"`
```
Comment (kflorianski-ibm) [inline api/v1/ibmlicensingdefinition_types.go:49]: This change should generate bundle changes, have you ran `make generate manifests bundle`?

**Review by kflorianski-ibm — CHANGES_REQUESTED** [review]

File: api/v1/ibmlicensingdefinition_types.go (hunk 2 above)
Comment (pgodowski) [reply api/v1/ibmlicensingdefinition_types.go:49]: Is this change safe, i.e. do we have preserve unknown fields so that this new field won't be pruned by k8s when older LS CRD is loaded into the cluster?

Also, CRD has to be regerenated, as Kacper wrote.

File: config/crd/bases/operator.ibm.com_ibmlicensingdefinitions.yaml [hunk 3]
```diff
@@ -77,6 +83,7 @@ spec:
             required:
             - action
             - condition
+            - remove
```
Comment (pgodowski) [inline config/crd/bases/operator.ibm.com_ibmlicensingdefinitions.yaml:86]: When extending CRD, one must not add a new requied field. Where it would fail is the following scenario:

- cluster has already some CRs of Kind: IBMLicensingDefinition, before a new version of LicSvc is installed
- such CR has some fields in, but not `remove` (since such field was not available in older version of License Service)
- License Service is updated, thus CRD definition is upgraded
- K8s will re-validate all the existing CRs in the cluster, against the updated CRD schema - in this case all the already existing CRs will fail such validation
- moreover, if someone wants to add a new IBMLicensingDefinition CR  using older schema when new LicSvc is installed, such operation also fail, since we require additional field.

Long story short: it is ok to add a new field, but not to mark it as required. Yet frankly, why do we actually require this new field in the first place? Perhaps we're just missing `omitempty`?

And then, given we now have both "set" and "remove" - should we actually say that both are optional?

**Review by pgodowski — CHANGES_REQUESTED** [review]

File: config/crd/bases/operator.ibm.com_ibmlicensingdefinitions.yaml (hunk 3 above)
Comment (kflorianski-ibm) [reply config/crd/bases/operator.ibm.com_ibmlicensingdefinitions.yaml:86]: I agree, both should now be optional. Do we want to enforce at least one is provided in the operand code?

Comment (alicenoknow) [reply config/crd/bases/operator.ibm.com_ibmlicensingdefinitions.yaml]: both fields are now optional, the operand will do the check for both set & remove empty during CR validation along with other checks and log `IbmLicensingDefinition CR <name> is incomplete and cannot be processed`

Comment (pgodowski) [reply config/crd/bases/operator.ibm.com_ibmlicensingdefinitions.yaml]: > log `IbmLicensingDefinition CR <name> is incomplete and cannot be processed`

OK, but pls add more verbose log message, i.e. indicate that one of `set` or `remove` is required - to make sure those who look at the logs are clear what is wrong and what to do / fix

**Review by kflorianski-ibm — APPROVED** [review]
